### PR TITLE
Epmlstrtai 705 Dtos and mappers for CaseController

### PR DESCRIPTION
**Pull request by glinko1567**
1. Created separate Case DTOs
2. Created Case DTO transformer (insted of Dozer)
3. Moved CaseController, Case DTOs and Case DTO Tranformer to com.epam.test_generator.controllers.caze package

### CODE REVIEW

File: src/main/java/com/epam/test_generator/services/CaseService.java [hunk 1]
```diff
@@ -153,44 +152,93 @@ public CaseDTO addCaseToSuit(Long projectId, Long suitId, EditCaseDTO editCaseDT
...
+                            editCaseDTO.getStatus(),
+                            editCaseDTO.getSteps(),
+                            editCaseDTO.getComment()
+                    );
+                    CaseDTO updatedCaseDTO = updateCase(projectId, suitId, editCaseDTO.getId(), updateCaseDTO)
+                            .getUpdatedCaseDto();
+                    updatedCases.add(updatedCaseDTO);
+                    break;
+                default:
+                    throw new BadRequestException("Wrong action argument");
```
Comment (aleksei-khitev) [inline src/main/java/com/epam/test_generator/services/CaseService.java]: It's better to add information about argument itself.
When you investigate an issue via logs it isn't enough to have such message.
You may do something like this:
`throw new BadRequestException(String.format("Wrong action argument: %s", editCaseDTO.getAction()));`

File: src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java [hunk 2]
```diff
@@ -331,64 +338,65 @@ public void addCase_RuntimeException_StatusInternalServerError() throws Exceptio
...
     }
 
     @Test
     public void updateCase_UpdateCase_StatusOk() throws Exception {
-        CaseUpdateDTO expectedDto = new CaseUpdateDTO(caseDTO, Arrays.asList(1L, 3L, 5L));
-        when(casesService.updateCase(SIMPLE_PROJECT_ID,SIMPLE_SUIT_ID, SIMPLE_CASE_ID, editCaseDTOList.get(0))).thenReturn(expectedDto);
+        CaseWithFailedStepsDTO expectedDto = new CaseWithFailedStepsDTO(caseDTO, Arrays.asList(1L, 3L, 5L));
+        when(casesService.updateCase(anyLong(), anyLong(), anyLong(), any(UpdateCaseDTO.class))).thenReturn(expectedDto);
+
+//        when(casesService.updateCase(SIMPLE_PROJECT_ID,SIMPLE_SUIT_ID, SIMPLE_CASE_ID, updateCaseDTO)).thenReturn(expectedDto);
```
Comment (aleksei-khitev) [inline src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java]: Do you have any reasons to save commented code?

File: src/main/java/com/epam/test_generator/config/JaversConfig.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
     /**
      * Config javers to connect to existing sql repository. Set Tag and Step classes as Value type
-     * for getting theirs copies in Case history.
+     * for getting theirs copies in caze history.
```
Comment (pomkine) [inline src/main/java/com/epam/test_generator/config/JaversConfig.java]: Why `caze` ?

File: src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        caseDTO.setName(caze.getName());
+        caseDTO.setDescription(caze.getDescription());
+        caseDTO.setSteps(new StepTransformer().toDtoList(caze.getSteps()));
+        caseDTO.setCreationDate(caze.getCreationDate());
+        caseDTO.setUpdateDate(caze.getUpdateDate());
+        caseDTO.setPriority(caze.getPriority());
+        caseDTO.setTags(caze.getTags().stream().map(new TagTransformer()::toDto).collect(Collectors.toSet()));
+        caseDTO.setStatus(caze.getStatus());
+        caseDTO.setComment(caze.getComment());
+        caseDTO.setJiraProjectKey(caze.getJiraProjectKey());
```
Comment (pomkine) [inline src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java]: Do we need jira related stuff on UI?

File: src/main/java/com/epam/test_generator/services/CaseService.java (hunk 1 above)
Comment (glinko1567) [reply src/main/java/com/epam/test_generator/services/CaseService.java]: Done.

File: src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java (hunk 2 above)
Comment (glinko1567) [reply src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java]: Fixed. Just a mistake.

File: src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java (hunk 4 above)
Comment (glinko1567) [reply src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java]: I forgot - did we decide not to return anything related to Jira to UI?

Comment (pomkine) [reply src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java]: Hm... Let's remove it for now since UI is not using it yet.

Comment (glinko1567) [reply src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java]: I forgot - did we decide not to send any Jira-related info to UI?

File: src/main/java/com/epam/test_generator/config/JaversConfig.java (hunk 3 above)
Comment (glinko1567) [reply src/main/java/com/epam/test_generator/config/JaversConfig.java]: Fixed

File: src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java (hunk 4 above)
Comment (glinko1567) [reply src/main/java/com/epam/test_generator/controllers/caze/CaseDTOsTransformer.java]: Ok. I removed all Jira-related fields from response.CaseDTO

**Pull request merged**